genienlp/data_utils/bootleg.py

```diff
@@ -0,0 +1,381 @@
+            elif 'actor' in title or 'actress' in title:
+                type = 'Q33999'
+            
+            elif 'language' in title or title in ['cinema of country or region']:
+                type = 'Q315'
+            
+            elif 'writer' in title or title in ['composer', 'screenwriter', 'editer', 'singer', 'businessperson',
+                                                'playwright',
+                                                'art collector', 'comedian', 'musician', 'aircraft pilot',
+                                                'philanthropist',
+                                                'restaurateur', 'guitarist', 'novelist', 'Wikimedia list article',
+                                                'Wikimedia disambiguation page', 'journalist', 'musical group']:
+                type = 'unk'
+        
+        
+        elif domain == 'music':
+            if title in ['song', 'single', 'musical composition', 'ballad', 'extended play', 'literary work',
+                         'television series', 'film', 'play']:
+                type = 'Q7366'
+            elif 'album' in title or title in []:
+                type = 'Q482994'
+            elif 'genre' in title or title in ['country', 'music by country or region', 'music term', 'republic',
+                                               'ethnic group', 'music scene', 'popular music', 'rock music',
+                                               'heavy metal', 'music', 'pop music', 'electronic music', 'music style']:
+                type = 'Q188451'
+            elif 'person' in title or 'actor' in title or 'musician' in title or \
+                    title in ['singer', 'musician', 'songwriter',
+                              'composer', 'producer',
+                              'singer-songwriter', 'musical group', 'drummer',
+                              'writer', 'philanthropist', 'public figure',
+                              'poet', 'guitarist', 'rapper', 'painter',
+                              'film director', 'dancer', 'screenwriter',
+                              'rock band', 'university teacher', 'journalist',
+                              'television presenter', 'film producer',
+                              'saxophonist', 'music pedagogue',
+                              'association football player', 'film score composer',
+                              'disc jockey', 'record producer', 'engineer',
+                              'human biblical figure', 'big band',
+                              'musical duo', 'girl group', 'entrepreneur',
+                              'boy band', 'musical ensemble', 'artist',
+                              'vocal group', 'heavy metal band',
+                              'literary character', 'lawyer', 'lyricist',
+                              'baseball player', 'pianist', 'recording artist',
+                              'autobiographer', 'fashion designer']:
+                type = 'Q5'
```

Comment:
This is just "human". It is not right in general. How did you choose these representative QIDs?

Reply:
As long as you are training single domain (or multi-domain like Mario but you only want NED for one domain) the representative QID shouldn't matter. However, I put a TODO for multidomain (e.g. SQA multidomain) where all types should be granular so film actor --> Q10798782 , film director -->  Q2526255, etc. instead of all mapping to Q5.

Reply:
Well, the representative QID matters when you extract the Bootleg type embedding right? Otherwise you wouldn't need a QID at all.

Reply:
Yeah, but we are not using bootleg embeddings anymore cause they didn't help. In fact, I removed all the code for integration level 2. 
(P.S. We can switch to "dump_preds" mode vs "dump_embeds" mode which saves tons of space!)